changes for timeInterval

hanbollar opened the pull request:
Fixes [#6164](https://github.com/AnalyticalGraphicsInc/cesium/issues/6164)

cesium-concierge commented:
![Signed CLA is on file.](https://img.shields.io/badge/CLA-signed-brightgreen.svg)

@hanbollar, thanks for the pull request! Maintainers, we have a signed CLA from @hanbollar, so you can review this at any time.

---

__I am a bot who helps you make Cesium awesome! [Contributions to my configuration are welcome.](https://github.com/AnalyticalGraphicsInc/cesium/tree/master/.concierge)__

:earth_africa: :earth_americas: :earth_asia:

hanbollar commented:
@hpinkos I updated the documentation and added in the throw statement. lmk if this is what you were looking for / if there's anything else you think I should do.

hanbollar commented:
@mramato updated to match that

ggetz commented:
Thanks @hanbollar, can you add a quick test as well?

ggetz commented on Specs/Core/TimeIntervalSpec.js:
```diff
@@ -100,6 +100,12 @@ defineSuite([
 
+    it('fromIso8601 throws error when given Invalid ISO 8601 date.', function() {
+        expect(function() {
+            expect(TimeInterval.fromIso8601({ iso8601 : '2020-08-29T00:00:00+00:00' }));
```
You don't need to wrap this in the second `expect`:

```
expect(function() {
    TimeInterval.fromIso8601({ iso8601 : '2020-08-29T00:00:00+00:00' });
});
```

hanbollar commented:
@ggetz removed second expect

ggetz commented:
Thanks @hanbollar !

hpinkos commented on Source/Core/TimeInterval.js:
```diff
@@ -143,6 +147,9 @@ define([
 
         var dates = options.iso8601.split('/');
+        if (dates.length !== 2) {
+            throw new DeveloperError('Invalid ISO 8601 date.');
```
`options.iso8601 is an invalid ISO 8601 interval`
